soc/arm/nxp_imx/rt/Kconfig.defconfig.series

```diff
@@ -75,73 +75,51 @@ config PM_MCUX_PMU
 
 endif # SOC_SERIES_IMX_RT10XX && PM
 
-if CODE_SEMC
+###########################################################################
 
-config FLASH_SIZE
-	default $(dt_node_reg_size_int,/memory@80000000,0,K)
-
-config FLASH_BASE_ADDRESS
-	default $(dt_node_reg_addr_hex,/memory@80000000)
-
-endif # CODE_SEMC
-
-if CODE_ITCM
-
-config FLASH_SIZE
-	default $(dt_node_reg_size_int,/soc/flexram@40028000/itcm@0,0,K) if SOC_SERIES_IMX_RT11XX
-	default $(dt_node_reg_size_int,/soc/flexram@400b0000/itcm@0,0,K) if SOC_SERIES_IMX_RT10XX
-
-config FLASH_BASE_ADDRESS
-	default $(dt_node_reg_addr_hex,/soc/flexram@40028000/itcm@0) if SOC_SERIES_IMX_RT11XX
-	default $(dt_node_reg_addr_hex,/soc/flexram@400b0000/itcm@0) if SOC_SERIES_IMX_RT10XX
-
-endif # CODE_ITCM
-
-if CODE_SRAM0
+DT_CHOSEN_RT_BOOT_DEV := rt,boot-device
```

Comment:
Yes, the overlay is on my todo list

The reason right now that I have the RT prefix is because I am specifically adding this to take advantage of a feature I know of in the RT family, and so currently I am only expecting it to be used on RT boards unless this is a common type of functionality that other vendors have on their parts as well, but my implementation here is very specific to RT using some kconfig tricks at the soc level. The RT prefix I was also imagining would help clarify that this feature is only possible because of something on the RT platform, which in this case is the RT ROM bootloader.